instructions/fault-injection/fault-injection.adoc

```diff
@@ -0,0 +1,150 @@
+= Chaos Testing
+include::_attributes.adoc[]
+
+Apply some chaos engineering by throwing in some HTTP errors or network delays. Understanding failure scenarios is a critical aspect of microservices architecture (aka distributed computing)
+
+[IMPORTANT]
+.Before Start
+====
+You should have only the following virtualservices and destinationrules in the `istio-tutorial` namespace
+
+[source, bash,subs="+macros,+attributes"]
+----
+$ oc -n istio-tutorial get destinationrule
+No resources found.
+
+$ oc -n istio-tutorial get virtualservice
+NAME       GATEWAYS             HOSTS   AGE
+customer   [customer-gateway]   [*]     18h
+----
+
+====
+
+[#503error]
+== HTTP Error 503
+
+By default, recommendation v1, v2 and v3 are being randomly load-balanced as that is the default behavior in Kubernetes/OpenShift
+
+[source,bash,subs="+macros,+attributes"]
+----
+$ oc -n istio-tutorial get pods -l app=recommendation
+NAME                                 READY   STATUS    RESTARTS   AGE
+recommendation-v1-7fbb8f794-ngw58    2/2     Running   0          21h
+recommendation-v2-77bc7bb94d-8wjxz   2/2     Running   0          17h
+recommendation-v3-5786cd744d-96q79   2/2     Running   0          17h
+
+$ while :; do curl http://${INGRESS_GATEWAY} ; done
+customer => preference => recommendation v1 from '77bc7bb94d-8wjxz': 782
+customer => preference => recommendation v1 from '5786cd744d-96q79': 734
+customer => preference => recommendation v1 from '7fbb8f794-ngw58': 1224
+customer => preference => recommendation v1 from '77bc7bb94d-8wjxz': 783
+customer => preference => recommendation v1 from '5786cd744d-96q79': 735
+customer => preference => recommendation v1 from '7fbb8f794-ngw58': 1225
+customer => preference => recommendation v1 from '77bc7bb94d-8wjxz': 784
+customer => preference => recommendation v1 from '5786cd744d-96q79': 736
+customer => preference => recommendation v1 from '7fbb8f794-ngw58': 1226
+----
+
+You can inject 503's, for approximately 50% of the requests
+
+[source,bash,subs="+macros,+attributes"]
+----
+$ oc -n istio-tutorial apply -f src/istiofiles/destination-rule-recommendation.yml
+destinationrule.networking.istio.io/recommendation created
+
+$ oc -n istio-tutorial apply -f src/istiofiles/virtual-service-recommendation-503.yml
+virtualservice.networking.istio.io/recommendation created
```

Comment:
Can we combine the two commands above into a single yaml file?

Reply:
No, because we need the destination rule for both the 503 and delay virtual services.

Reply:
It's still simpler to have everything in one file per section then it's just a single create/delete and we shouldn't have to deal with people getting things wrong